accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,790 @@
+| net5.0-watchos  | xamarin.watchos                                            |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-windows  | netcoreapp1..3.1                                           | WinForms + WPF                    |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| Tizen, Unity... | Will follow the Xamarin model                              |                                   |
+
+_**Open Issue**. In case of Xamarin, there is a discussion of whether the
+convergence with .NET Core will be a breaking change for existing NuGet packages
+(for example, there is a desire to change their namespaces). In that case, we
+should not map it to the existing TFM as that would be pointless._
+
+_**Open Issue**. Will the above scheme work for iPad?_
+
+### OS versions
+
+The TFM will support an optional OS version, such as:
+
+* `net5.0-ios`
+* `net5.0-ios12.0`
+
+A TFM without an OS version will be interpreted as the lowest OS version that
+was supported by the corresponding `netX.Y` version.
+
+> *Note: One assumption here is that developers can always compile for the
+> latest OS API set and run on older OS versions so long they guard method calls
+> correctly. This is how P/Invokes, WinRT, iOS, and Android bindings work
+> today.*
+
+For example, say that when .NET 5 ships, the version of iOS that we include
+bindings for is iOS 13. That would mean that `net5.0-ios` and `net5.0-ios13.0`
+mean the exact same thing. Now lets say that Apple ships a new iOS 14 before
+.NET 6 is ready. We'd release a new version of the .NET 5 SDK that adds support
+for iOS 14. On machines with that SDK `net5.0-ios` still means `net5.0-ios13.0`.
+
+Please note that this mapping is specific for `net5.0`. When .NET 6 ships, we
+could decide that `net6.0-ios` means `net6.0-ios14.0`. But even when you use the
+.NET 6 SDK, when targeting .NET 5, `net5.0-ios` also still means
+`net5.0-ios13.0`. So these mappings are immutable.
+
+We have done this to simplify the experience for application models where the OS
+version is largely irrelevant, for example WinForms and WPF. Whether or not the
+template will put an OS version in the `<TargetFramework>` is up to each
+application model. Based on conversations, it seems we'll be landing on this:
+
+| TFM             | Project file includes OS version
+|-----------------|---------------------------------
+| net5.0-android  | Yes
+| net5.0-ios      | Yes
+| net5.0-macos    | Yes
+| net5.0-tvos     | Yes
+| net5.0-watchos  | Yes
+| net5.0-windows  | No
+
+Please note that by being able to put the OS version into the TFM one can also
+multi-target between OS versions:
+
+```XML
+<PropertyGroup>
+  <TargetFrameworks>net5.0-ios13.0;net5.0-ios14.0</TargetFrameworks>
```

Comment:
This a combination of both `TargetFramework` which is `net5.0` and `TargetPlatform` which is `ios14.0`.

It is no longer relevant to call it just TargetFrameworks, we should opt for a simpler name, e.g.: `BuildTargets`/`TargetProfiles`/`Targets`, etc... I'm not good at naming things, sorry.

Reply:
A "profile" has a very specific meaning, so we can't use that.

> This a combination of both `TargetFramework` which is `net5.0` and `TargetPlatform` which is `ios14.0`.

The point of this proposal is to add platform information to the TFM friendly name so that it's easier to talk about. This isn't a bug, that's the feature! 😄

Reply:
I get what it represents but... 

When I was talking about this to my colleagues who were either windows guys or new to .NET, they kept asking about how it was an extern call to platform APIs and not something that's already in the .NET BCLs or a superset (of BCLs) like you mentioned.

**They were confused**. When I told them to think it as a separate lib that's present only on that platform (indicated by platform id/version, like WinRT), they were in the clear.

Before that, they were asking all sorts of Qs about versioning and compatability but after mentioning WinRT, they sorta understood.